### PR TITLE
`Development`: Remove all temporary files when using the ProgrammingExerciseTestService

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
I noticed that running the server tests create a lot of temporary files which never got cleaned up. These files got created by the `ProgrammingExerciseTestService` and are supposed to get deleted when calling `tearDown()`. However, some tests only called `setup()` (which creates these files) and missed calling `tearDown()`. 

### Description
I added the missing method calls.

### Steps for Testing
code review

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

**Review by DominikRemo — APPROVED** [review]
Thank you for this improvement :)
Code LGTM and tests still pass on Bamboo and GitHub Actions

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

File: src/test/java/de/tum/in/www1/artemis/util/GitUtilService.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class GitUtilService {
     // Note: the first string has to be same as artemis.repo-clone-path (see src/test/resources/config/application-artemis.yml) because here local git repos will be cloned
     private final Path localPath = Path.of(".", "repos", "server-integration-test").resolve("test-repository").normalize();
 
-    private final Path remotePath = Files.createTempDirectory("remotegittest").resolve("scm/test-repository");
+    private final Path remotePath = Path.of(System.getProperty("java.io.tmpdir")).resolve("remotegittest/scm/test-repository");
```
Comment (pal03377) [inline src/test/java/de/tum/in/www1/artemis/util/GitUtilService.java:33]: Just a question (the change is fine by me): Why is this change necessary? Shouldn't it work with the code before as well?

**Review by pal03377 — APPROVED** [review]
Good improvement, thanks! I only have one non-blocking question.

File: src/test/java/de/tum/in/www1/artemis/util/GitUtilService.java (hunk 1 above)
Comment (Strohgelaender) [reply src/test/java/de/tum/in/www1/artemis/util/GitUtilService.java:33]: `createTempDirectory` creates a new temporary directory every time this service gets instantiated (i.e. every time a local test run gets started) even if no methods on this service are called. This means that a new folder gets created, but never deleted. Not we just save the path here and create the folder only when calling `initRepo`, which means that the folder also gets deleted when later calling `deleteRepos`
